### ISSUE TITLE
Human readable files/folders as Genus_species_strain

### ISSUE THREAD

**Issue by tseemann** [kblin/ncbi-genome-download#8]
In my (older unreleased scruffier) version of this tool I create a hierarchy like this:

```
Kingdom/
   Genus/
       species/
            strain/
                    blah.gbk
                    blah.fna
                    blah.gff
```

I currently set `blah` to `Genus_species_strain` but that loses the `GCA_xxxx` accession. I was thinking of having a a 'mirror' folder of symlinks with human readable names.

It was tricky to extract the `strain` as it appears in up to 3 different columns sometimes, but it mostly works.

The reason for this is to make it easy to work with sequences and get human readable labels etc.

**Comment by kblin** [comment]
What is your logic to extract the strain name?

**Comment by tseemann** [comment]
FIrst I try and match the strain from column 8
` $x[7] =~ m/^(\S+)\s+(\S+)(\s+.*)?$/`

Then if that fails I try column 9:
`$x[8] =~ m/^strain=(.*)$/)`

If that fails I use the assembly ID:
`$strain ||= $x[0];`

And then i sanitize the string:

`  $s =~ s/[\[\]]/ /g;
  $s =~ s/^\s+//g;
  $s =~ s/\s+$//g;
  $s =~ s/\s+/_/g;
  $s =~ s/['"]/_/g;
  $s =~ s{[/;:]}{-}g;
  $s =~ s/_+/_/g;`

**Comment by kblin** [comment]
Ok, so I guess my current strategy would be:

```python
def find_strain(entry):
    """entry is a dictionary parsed from assembly_summary.txt,
        with keys equal to the column headers"""
    strain = entry['infraspecific_name'].split('=')[-1]
    if strain != '':
        return strain
    if strain != '':
        return strain
    if len(entry['organism_name'].split(' ')) > 2:
        strain = ' '.join(entry['organism_name'].split(' ')[2:])
        return strain
    return entry['assembly_accession']
```

plus maybe some extra cleanup like @tseemann described.

However, prior to Python 3.2 `os.symlink` is just available on Unix systems, even when running on Windows versions that support symbolic links. I'm not sure I want to start adding system-dependent code into `ncbi-genome-download`. I'll have to think about this for a bit.

**Comment by tseemann** [comment]
Ok, i'll poke you in a few months, thanks!

**Comment by kblin** [comment]
So, apart from making up my mind how to best do the cleanup, I've got a working implementation of this. Expect this in the next version of `ncbi-genome-download`.

**Comment by kblin** [comment]
Fixed by 10e38ba, and in the 0.2.0 release.
